File: examples-sdk/typescript/005-basic-multiservice-mesh-ndjson.ts
#!/usr/bin/env tsx
// 005-basic-multiservice-mesh-ndjson — TypeScript SDK equivalent
// Compile: vil compile --from typescript --input 005-basic-multiservice-mesh-ndjson.ts --release

import { VilPipeline, VilServer, ServiceProcess } from "vil-sdk";

const p = new VilPipeline("MultiServiceMesh", 3084);
p.sink({ port: 3084, path: "/ingest", name: "gateway" });
p.source({ name: "credit_ingest", format: "json" });
p.route("gateway.trigger_out", "ingest.trigger_in", "LoanWrite");
p.route("ingest.response_data_out", "gateway.response_data_in", "LoanWrite");
p.route("ingest.response_ctrl_out", "gateway.response_ctrl_in", "Copy");
p.compile();

File: examples-sdk/python/005-basic-multiservice-mesh-ndjson.py
#!/usr/bin/env python3
"""005-basic-multiservice-mesh-ndjson — Python SDK equivalent
Compile: vil compile --from python --input 005-basic-multiservice-mesh-ndjson.py --release
"""
import os
from vil import VilPipeline, VilServer, ServiceProcess

pipeline = VilPipeline("MultiServiceMesh", 3084)
pipeline.sink(port=3084, path="/ingest", name="gateway")
pipeline.source(format="json", name="credit_ingest")
pipeline.route("gateway.trigger_out", "ingest.trigger_in", "LoanWrite")
pipeline.route("ingest.response_data_out", "gateway.response_data_in", "LoanWrite")
pipeline.route("ingest.response_ctrl_out", "gateway.response_ctrl_in", "Copy")
pipeline.compile()
